### PR TITLE
Replaced old repository with new one. Addded liquibase support. Imple…

### PR DESCRIPTION
**Pull request by Trayvor**
…mented safe delete concept for Book entity. Added new endpoints to BookController and BookService

### CODE REVIEW

File: .jpb/jpb-settings.xml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  <component name="PersistenceUnitSettings">
+    <persistence-units>
+      <persistence-unit name="Default">
+        <packages>
+          <package value="org.example.bookstore" />
+        </packages>
+      </persistence-unit>
+    </persistence-units>
+  </component>
+</project>
```
Comment (MentorMateAcademy) [inline .jpb/jpb-settings.xml]: Is it mandatory?

**Review by MentorMateAcademy — CHANGES_REQUESTED** [review]

File: src/main/java/org/example/bookstore/model/Book.java [hunk 2]
```diff
@@ -28,5 +32,8 @@ public class Book {
     private BigDecimal price;
     @Column(nullable = false)
     private String description;
+    @Column(name = "cover_image")
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookstore/model/Book.java]:
```suggestion
```

File: src/main/java/org/example/bookstore/service/impl/BookServiceImpl.java [hunk 3]
```diff
@@ -33,8 +34,21 @@ public List<BookDto> findAll() {
...
-        return bookMapper.toDto(bookRepository.getById(id).orElseThrow(
+        return bookMapper.toDto(bookRepository.findById(id).orElseThrow(
                 () -> new EntityNotFoundException("Cant find book with id " + id)
         ));
     }
+
+    @Override
+    public BookDto updateById(Long id, UpdateBookRequestDto requestDto) {
+        Book model = bookMapper.toModel(requestDto);
+        model.setId(id);
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookstore/service/impl/BookServiceImpl.java]:
```suggestion
        Book book = bookRepository.findById(id)
                .orElseThrow(() ....
        bookMapper.updateBookFromDto(bookDto, book);
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Good job! let's improve it

File: src/main/java/org/example/bookstore/service/impl/BookServiceImpl.java (hunk 3 above)
Comment (Trayvor) [reply src/main/java/org/example/bookstore/service/impl/BookServiceImpl.java:45]: what is incorrect here?

**Review by Elena-Bruyako — APPROVED** [review]

**Pull request merged**